### PR TITLE
revert log driver to use json-file, and rotate

### PR DESCRIPTION
**Pull request by rjtsdl**
**What this PR does / why we need it**:
1. Revert back to use json-file as docker log driver
2. Set log rotate policy by leveraging docker log-opt
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
#689 
**Special notes for your reviewer**:
Ready to review
**Release note**:

```release-note
There are many people bothered by the fact we switch to journald for logging, because the change break their existing logging setup. This change is to change back to the user familiar behavior, and also handled container logs through docker log-opt. It should also address the "disk filled up with log" issue. 
```

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/766)

### CODE REVIEW

File: parts/kubernetesmastercustomdata.yml [hunk 1]
```diff
@@ -19,25 +19,19 @@ write_files:
...
-    RateLimitInterval=1s
-    RateLimitBurst=20000
-  
 - path: "/etc/docker/daemon.json"
   permissions: "0644"
   owner: "root"
   content: |
     {
-      "live-restore": true
+      "live-restore": true,
```
Comment (anhowe) [inline parts/kubernetesmastercustomdata.yml]: does this do logrotate?

Comment (rjtsdl) [inline parts/kubernetesmastercustomdata.yml]: I added the things under this line. This exists before my change.

Comment (seanknox) [inline parts/kubernetesmastercustomdata.yml]: @anhowe The settings below will constrain Docker logs to 50m, with a max of 5 files. 👍

**Comment by rjtsdl** [comment]
I was testing on one of the master machines, here is what i saw.
It rotates with 5 files with 50M upper limit file size. 

So it looks good.

```
root@k8s-master-24463227-0:/var/lib/docker/containers/a69f8412fd11f163f6e0df34f12968661f2ab2146a12bc19ee1132f2bef86870# ls -lh
total 238M
-r-------- 1 root root  47M Jun 14 19:09 a69f8412fd11f163f6e0df34f12968661f2ab2146a12bc19ee1132f2bef86870-json.log
-r-------- 1 root root  48M Jun 14 19:09 a69f8412fd11f163f6e0df34f12968661f2ab2146a12bc19ee1132f2bef86870-json.log.1
-r-------- 1 root root  48M Jun 14 19:09 a69f8412fd11f163f6e0df34f12968661f2ab2146a12bc19ee1132f2bef86870-json.log.2
-r-------- 1 root root  48M Jun 14 19:09 a69f8412fd11f163f6e0df34f12968661f2ab2146a12bc19ee1132f2bef86870-json.log.3
-r-------- 1 root root  48M Jun 14 19:09 a69f8412fd11f163f6e0df34f12968661f2ab2146a12bc19ee1132f2bef86870-json.log.4
-rw-rw-rw- 1 root root 2.5K Jun 14 18:57 config.v2.json
-rw-rw-rw- 1 root root 1.1K Jun 14 18:57 hostconfig.json
-rw-r--r-- 1 root root   13 Jun 14 18:57 hostname
-rw-r--r-- 1 root root  174 Jun 14 18:57 hosts
-rw-r--r-- 1 root root  235 Jun 14 18:57 resolv.conf
-rw-r--r-- 1 root root   71 Jun 14 18:57 resolv.conf.hash
drwxrwxrwt 2 root root   40 Jun 14 18:57 shm
```

**Comment by sjdweb** [comment]
Hi, I have a cluster created last week that has journald still enabled. How can I update this cluster to match `v0.2.0` release? Can't find any documentation about updating existing clusters. Thanks

**Pull request merged**